tests/demo_tests/test_ip_v4v6_raw_capture.py

```diff
@@ -0,0 +1,196 @@
+import pytest
+import snappi
+
+
+@pytest.fixture
+def app_intf(utils):
+    api = snappi.api(host=utils.settings.api_server, ext=utils.settings.ext)
+    yield api
+    if getattr(api, 'assistant', None) is not None:
+        api.assistant.Session.remove()
+
+
+def test_ip_v4v6_raw_capture(app_intf, utils):
+    """
+    Test to demonstrate ipv4 and ipv6 device and raw type traffic
+    configuration, transmit, statistics and capture.
+    """
+    config = app_intf.config()
+
+    size = 256
+    packets = 1000
+    src = '00:10:10:20:20:10'
+    dst = '00:10:10:20:20:20'
+    mac_step = '00:00:00:00:00:01'
+
+    src_ipv4 = '10.1.1.1'
+    dst_ipv4 = '10.1.1.2'
+    ipv4_step = '0.0.1.0'
+
+    src_ipv6 = 'abcd::1a'
+    dst_ipv6 = 'abcd::2a'
+    ipv6_step = '1::'
+
+    ##############################
+    # Ports configuration
+    ##############################
+    tx, rx = (
+        config.ports
+        .port(name='tx', location=utils.settings.ports[0])
+        .port(name='rx', location=utils.settings.ports[1])
+    )
+    l1 = config.layer1.layer1()[-1]
+    l1.name = 'L1 Settings'
+    l1.port_names = [tx.name, rx.name]
+    l1.speed = utils.settings.speed
+    l1.media = utils.settings.media
+
+    cap = config.captures.capture(name='c1')[-1]
+    cap.port_names = [rx.name]
+    cap.format = cap.PCAP
+
+    ##############################
+    # Flows configuration
+    ##############################
+    f1, f2 = (
+        config.flows
+        .flow(name='FlowIpv4Raw')
+        .flow(name='FlowIpv6Raw')
+    )
+
+    f1.tx_rx.port.tx_name = tx.name
+    f1.tx_rx.port.rx_name = rx.name
+    f1.packet.ethernet().ipv4().tcp()
+    eth = f1.packet[0]
+    ipv4 = f1.packet[1]
+    tcp = f1.packet[-1]
+    eth.src.increment.start = src
+    eth.src.increment.step = mac_step
+    eth.src.increment.count = 10
+    eth.dst.decrement.start = dst
+    eth.dst.decrement.step = mac_step
+    eth.dst.decrement.count = 10
+    ipv4.src.increment.start = src_ipv4
```

Comment:
add a newline with single line comment for every packet header (e.g. before ether, ip, tcp)

Reply:
addressed